### PR TITLE
feat: allow custom headers to be sent through http requests

### PR DESCRIPTION
**Pull request by tanguyantoine**
# What 

- add new options to CLI: extra_http_headers

# Why

- it allows to provide extra headers to requests sent to Metabase instance

# Use case

it allows instances to be reached when they are deployed behind security system (ex https://developers.cloudflare.com/cloudflare-one/identity/service-tokens/#connect-your-service-to-access)

example

```bash
dbt-metabase models --http_extra_headers CF-Access-Client-Id $CF_CLIENT_ID --http_extra_headers CF-Access-Client-Secret $CF_CLIENT_SECRET
```  

thanks to @elongl and @elementary-data for the initial implementation

https://github.com/elementary-data/dbt-metabase/commit/91c34651ab323653c993820222670c27df19b38f

### CODE REVIEW

**Comment by tanguyantoine** [comment]
@gouline I have run black to format the files correctly

**Comment by gouline** [comment]
To save time, run all the checks locally prior to pushing commits.

**Comment by tanguyantoine** [comment]
@gouline sorry for that, I have run make check and it is working now :-)

File: dbtmetabase/__init__.py [hunk 1]
```diff
@@ -86,7 +86,8 @@ class OptionAcceptableFromConfig(click.Option):
     more resilence to raising an error when the option exists in the users config.
 
     This also overrides default values for boolean CLI flags (e.g. --use_metabase_http/--use_metabase_https) in options when
-    no CLI flag is passed, but a value is provided in the config file (e.g. metabase_use_http: True)."""
+    no CLI flag is passed, but a value is provided in the config file (e.g. metabase_use_http: True).
```
Comment (gouline) [inline dbtmetabase/__init__.py:89]: Revert all changes unrelated to the functionality you're implementing (e.g. removing newlines, moving comment quotes)

**Review by gouline — CHANGES_REQUESTED** [review]

File: dbtmetabase/__init__.py (hunk 1 above)
Comment (gouline) [reply dbtmetabase/__init__.py]: Sorry, I discovered this was misalignment in formatter versions, which I have now fixed in the master.

**Review by gouline — APPROVED** [review]

**Pull request merged**